### PR TITLE
chore: remove dependency shelljs

### PR DESCRIPTION
**Pull request by fasttime**
This PR removes the dependency shelljs and replaces its usages with native Node.js API calls in unit tests.

ShellJS in versions prior to 0.8.5 has a high rated vulnerability: https://nvd.nist.gov/vuln/detail/CVE-2022-0144.

Most Shelljs functions are synchronous, but for Node.js I/O operations it is recommended to use asynchronous functions, so I decided to use `fs/promises`. We could also switch to the synchronous versions of those functions to stay in line with the usage in other parts of this repo.

### CODE REVIEW

File: tests/init/config-initializer.js [hunk 1]
```diff
@@ -115,8 +115,8 @@ describe("configInitializer", () => {
         npmFetchPeerDependenciesStub.resetHistory();
     });
 
-    after(() => {
-        sh.rm("-r", fixtureDir);
+    after(async () => {
+        await fsp.rm(path.join(fixtureDir, "../../.."), { recursive: true });
```
Comment (fasttime) [inline tests/init/config-initializer.js:119]: This should also remove "tmp" in the current directory, not just "tmp/eslint/fixtures/config-initializer".

**Review by nzakas — APPROVED** [review]
LGTM. Would like @aladdin-add to review before merging.

File: tests/init/config-initializer.js [hunk 2]
```diff
@@ -8,11 +8,11 @@
 // Requirements
 //------------------------------------------------------------------------------
 
+import fsp from "node:fs/promises";
```
Comment (aladdin-add) [inline tests/init/config-initializer.js:11]: We're rewriting it in [another PR](https://github.com/eslint/create-config/pull/81), and this file will be removed.

I can remove `shelljs` in that PR, thoughts?

Comment (fasttime) [reply tests/init/config-initializer.js:11]: Sounds good 👍🏻 It looks like shelljs isn't used at all in that branch, so you could simply remove the package.json dependency.

Comment (aladdin-add) [reply tests/init/config-initializer.js:11]: done in 46027138738a8296a39dfd21b0221e36d9d1f805.

**Pull request closed without merging**